src/util/cstr/fd_cstr.c

```diff
@@ -24,6 +24,27 @@ double       fd_cstr_to_double( char const * cstr ) { return         strtod ( cs
 
 ulong fd_cstr_to_ulong_octal( char const * cstr ) { return (ulong)strtoul( cstr, NULL, 8 ); }
 
+#if FD_HAS_HOSTED
+/* TODO: Provide a non-hosted implementation */
+ulong
+fd_cstr_to_ip4_addr( char const * s ) {
+  int  n=0;
+  uint x[4];
+
+  int res = sscanf( s, "%u.%u.%u.%u%n", &x[0], &x[1], &x[2], &x[3], &n );
+
+  if( FD_UNLIKELY( res!=4
+                || n<0 || s[n]!='\0'
+                || x[0]>UCHAR_MAX
+                || x[1]>UCHAR_MAX
+                || x[2]>UCHAR_MAX
+                || x[3]>UCHAR_MAX ) )
+    return ULONG_MAX;
```

Comment:
Just thinking out loud: wonder if it might be worth defining a `FD_FN_CONST static inline is_valid_ip4_addr` function which just returns false if the parameter is `ULONG_MAX`. That way the user doesn't have to remember the convention used for representing invalid values.